added shortcuts/hotkeys for tileset editor plugin collision buttons

GryphonClaw opened the pull request:
Adds hotkeys to the tileset editor collision buttons

default shortcuts:
New Rectangle Collision SHIFT-R
New Polygon Collision SHIFT-P
Delete selected collision SHIFT-BACKSPACE (delete on mac)

Calinou committed:
Require Ctrl for switching between editors, bind F2 to Rename Node

Switching between editors now requires holding Ctrl to avoid
conflicts with the new F2 shortcut.

The asset library can now be accessed by pressing Ctrl + F4
on Windows and Linux, or Alt + 4 on macOS.

This partially addresses #38139.

Calinou committed:
Improve the 3D editor selection box appearance

- Draw two boxes slightly offset from each other to give the illustion
  of a thicker outline.
- Decrease the offset compared to the 3D node's AABB to give a more
  accurate representation of its size.
- Make the box fully visible instead of only displaying the corners.
- Draw a x-ray version of the box that's more translucent, but visible
  through walls. This helps make the box more visible while still
  having a sense of depth.
- Use an orange color similar to the 2D editor.

marstaik committed:
Alpha Hash and Alpha2Coverage Implementation

Chaosus committed:
Exposed randi_range to global funcs + renamed rand_range to randf_range

RandomShaper committed:
Improve/fix packed data API

- Enhance directory API
- Fix `FileAccess::exists()` not checking for PackedData being disabled
- Fix moving to the parent directory (`..`)
- Allow absolute paths in existence checks

Calinou committed:
Improve messages related to overriding the default editor layout

This closes #33884.

Meriipu committed:
If the mouse is held on notification_wm_mouse_exit, do not drop focus

This fixes a bug where users of the scrollbar had to be very careful
not to move the mouse outside the viewport, otherwise the scrollbar
would drop its drag-action and stop scrolling until clicked again.

The existing behaviour had the side-effect of also dropping the
cosmetic highlighting of the scrollbar (in addition to the dragging),
for the specific case where the mouse was move outside the window.
The previous behaviour did nothing to remove the highlight if the
mouse was released (but not moved) inside the viewport.

This separate issue with the lingering highlight of the scrollbar
(until a mouse-movement action is performed inside the viewport) is
fixed in an immediate followup to this commit.

Closes bug #39634

Meriipu committed:
Make the currently hovered control get updated on mouse-release

Previously, when the mouse was released after dragging a scrollbar,
its highlight was not dropped (if the mouse cursor was still inside
the viewport). This seems to be because the currently hovered control
only gets updated when the mouse is moved.

This commit fixes the dropping of the cosmetic highlight by running
the check for whether the currently hovered control has changed on
mouse-clicks, in addition to to the existing mouse-movements.

naithar committed:
iOS Modules: separate main platform code from modules

Moved previously builtin modules 'GameCenter', 'AppStore', 'iCloud' to separate modules to be represented as plugin.
Modified 'ARKit' and 'Camera' to not be builtin into engine and work as plugin.
Changed platform code so it's not affected by the move.
Modified Xcode project file to remove parameters that doesn't make any effect.
Added basic '.gdip' plugin config file.

naithar committed:
iOS: change platform code to support plugins

naithar committed:
iOS Export: export modification to support plugins

Added plugin configuration.
Export options now use plugins that could be enabled/disabled.
Plugin changes are observed at runtime.

Calinou committed:
Rename the "Delete" option in the FileSystem dock to "Move to Trash"

It actually moves files to the system trash instead of removing them
completely.

naithar committed:
iOS Export: support multi-target plugin

Plugins can use 'binary_name.a' or 'binary_name.release.a' and 'binary_name.debug.a' for plugin library.

Calinou committed:
Remove `debug_symbols=full` in favor of `debug_symbols=yes`

`debug_symbols=yes` will now behave like `debug_symbols=full` did
before. The difference in compressed file sizes is not that large,
which means there isn't much point in having two different values.

This helps make the buildsystem easier to understand.

Lunatoid committed:
Fixed ParseError when calling Object.new()

Fixes #41462 where calling Object.new() in GDScript gave an error.
I fixed it by adding exclusion when checking if the name is a builtin
type to exclude objects with a comment detailing why.

Calinou committed:
Improve the documentation related to fullscreen and borderless settings

EricEzaM committed:
Added hysteresis for popup sub-menus

This adds a small lag effect when opening submenus which allow the user to move directly to an item on the submenu without worrying about avoiding the autohide regions.

Calinou committed:
Add a test suite for ConfigFile

Calinou committed:
Rename the `type` parameter to `node_type` in Theme and Control

This makes it clearer that it expects a node type as a string
(such as "Label") instead of a type like "TYPE_ARRAY".
This is backwards-compatible since only the name of the parameter
is changed, not its order.

HaSa1002 committed:
Port code examples to C# (D)

Includes:
 * Decal
 * Dictionary
 * Directory
 * DisplayServer
 * DTLSServer
 * DynamicFont
 * EditorImportPlugin
 * EditorPlugin
 * EditorScenePostImport
 * EditorScript
 * EditorSettings
 * EditorTranslationParserPlugin
 * Engine
 * Expression

Co-authored-by: Aaron Franke <arnfranke@yahoo.com>

madmiraal committed:
Remove the unnecessary sync() and the restrictions it imposes on 3D Physics.

aaronfranke committed:
Add line removal around braces to the file formatting script

aaronfranke committed:
Remove empty lines around braces with the formatting script

Xrayez committed:
Describe `ImageTexture`, `Image` creation and usage

aaronfranke committed:
Allow creating projects in non-empty folders with a confirmation popup

aaronfranke committed:
Improve comments in Color documentation

opl- committed:
Improve output in Debugger Errors tab for scripts

Calinou committed:
Add files to create a Windows editor installer using Inno Setup

This partially addresses
https://github.com/godotengine/godot-proposals/issues/1432.

To fully address the proposal above, official Windows installers will
have to be compiled and distributed.

theoway committed:
Fixes the misleading error message for call_recursive method for TreeItems

Calinou committed:
Document the supported TLS versions in HTTPRequest

See #42857.

Calinou committed:
Increase the default HTTPClient download chunk size to 64 KiB

This improves download speeds at the cost of increased memory usage.

This change also effects HTTPRequest automatically.

See #32807 and #33862.

Calinou committed:
Add `Image.load_bmp_from_buffer()` for run-time BMP image loading

This partially addresses
https://github.com/godotengine/godot-proposals/issues/676.

Calinou committed:
Allow more image types in the roughness texture's Src Normal import hint

erictuvesson committed:
fix(editor): TileMap floodfill with same tile ID and different variation

Closes https://github.com/godotengine/godot/issues/40435

rsubtil committed:
Changed path behaviour for Windows

Faless committed:
[HTML5] Port JavaScript inline code to libraries.

The API is implemented in javascript, and generates C functions that can
be called from godot.
This allows much cleaner code replacing all `EM_ASM` calls in our C++
code with plain C function calls.
This also gets rid of few hacks and comes with few optimizations (e.g.
custom cursor shapes should be much faster now).

Faless committed:
RasterizerDummy fixes.

aaronfranke committed:
Add raw strength value for internal use

Faless committed:
[HTML5] Update syntax for lto.

Faless committed:
Fix gdnative build when WebRTC module is disabled.

vnen committed:
GDScript: Fix native class not set with inheritance

bruvzg committed:
[iOS] Add missing ARC flag to the simulator build.

YeldhamDev committed:
Add icons for the 'CanvasGroup' and 'CodeEdit' nodes

madmiraal committed:
Remove unneeded filter on joy_axis()

HaSa1002 committed:
MainLoop: Remove deprecated methods

HaSa1002 committed:
Docs: Port Code Examples to C# (M, N, O, P, R)

Includes:
 * MarginContainer
 * NavigationPolygon
 * Node
 * NodePath
 * OS
 * PackedByteArray
 * PackedScene
 * PacketPeerUDP
 * PCKPacker
 * Performance
 * PhysicsShapeQueryParameters2D
 * PhysicsShapeQueryParameters3D
 * PrimitiveMesh
 * ProjectSettings

Co-authored-by: Aaron Franke <arnfranke@yahoo.com>

HEAVYPOLY committed:
Change android orientations from sensor to user

# Conflicts:
#	platform/android/java/lib/src/org/godotengine/godot/GodotIO.java

HaSa1002 committed:
Docs: MeshDataTool: showcase tool in code example

aaronfranke committed:
Add LStrip and RStrip to C# strings

aaronfranke committed:
Add HexEncode to C#

Calinou committed:
Correct documentation about NinePatchRect stretch modes

This closes https://github.com/godotengine/godot-docs/issues/4333.

TheCire committed:
added shortcuts/hotkeys for tileset editor plugin collision buttons

akien-mga committed:
Tooltips: Improve code clarity and docs

The return type for `_make_custom_tooltip` is clarified as Control, and users
should make sure to return a visible node for proper size calculations.

Moreover in the current master branch, a PopupPanel will be added as parent
to the provided tooltip to make it a sub-window.

Clarifies documentation for `Control._make_custom_tooltip`, and shows how to
use the (until now undocumented) "TooltipPanel" and "TooltipLabel" theme types
to style tooltips.

Fixes #39677.

Calinou committed:
Remove `Color.contrasted()` as its behavior is barely useful

Returning the most contrasting color isn't a trivial task, as there
are often many possible choices. It's usually best left for the user
to implement using a script.

KoBeWi committed:
Release pressed action if event is removed

reduz committed:
Implement DirectionalLight2D

Also separated Light2D in PointLight2D and DirectionalLight2D.
Used PointLight2D because its more of a point, and it does not work
the same as OmniLight (as shape depends on texture).
Added a few utility methods to Rect2D I needed.

rcorre committed:
Fix typo in Tabs doc.

I don't know enough about grammar to say why the "ing" suffix is
correct, but I'm reasonably sure it is :)

aaronfranke committed:
Move call to update the 3D grid into check for updating the view

akien-mga committed:
doc: Sync classref with current source + fixup some bindings

Includes various changes triggered by the refactoring of method bindings.

YeldhamDev committed:
Fix WAV resources ignoring the AudioServer's 'global_rate_scale' value

akien-mga committed:
doc: Override default value for RandomNumberGenerator.seed

It's non-deterministic so it's better to show a fixed value like 0 instead of
having it potentially change whenever `randomize()` is called.

Fixes #43317.

reduz committed:
Refactored Variant Operators.

-Using classes to call and a table
-For typed code (GDS or GDNative), can obtain functions to call prevalidated or ptr.

YeldhamDev committed:
Disallow setting the AudioServer's 'global_rate_scale' to a value equal or inferior to 0

Commit:
Remove unused `PHI` define in math funcs

bruvzg committed:
Fix uninitialised variables in the BaseMaterial3D.

Commit:
Fix completion for built-in load function

madmiraal committed:
Use LONG instead of DWORD (unsigned int) when interacting with DIJOFS constants.

KoBeWi committed:
Add docs for editor viewport drawing methods

KoBeWi committed:
Register methods for drawing 3D editor overlays

Faless committed:
[HTML5] Fix audio buffer size and latency hint.

The size of the audio buffer was incorrectly doubled when creating the
script processor.
latencyHint is expressed in seconds, not milliseconds.
Additionally, on some browsers it actually affect the performance and
stability of the audio driver.
For this reason it has been completely disabled (interactive) and a not
has been left for future reference.

Calinou committed:
Fix "unifom" typo in RasterizerStorageRD method

Calinou committed:
Allow using SVG images as project icon

Since exporters will save their own icon, the target platforms
don't have to support SVG to display the icon correctly.

This closes #23068.

reduz committed:
Refactored variant setters/getters

-Discern between named, indexed and keyed
-Get direct access to functions for typed GDScript and GDNative bindings
-Small changes to some classes in order to work with the new setget binder

Chaosus committed:
[Mono] Added Shuffle method to Array

YeldhamDev committed:
Add icons for the new Light2D nodes

vnen committed:
GDNative: Define special char types in string.h

Those are standard types in C++ but not in C.

This also removes the wchar header which is not needed anymore and use
stddef.h instead (which is needed for size_t).

reduz committed:
Reorganized core/ directory, it was too fatty already

-Removed FuncRef, since Callable makes it obsolete
-Removed int_types.h as its obsolete in c++11+
-Changed color names code

Calinou committed:
Use Ctrl+Shift+C as the default FileSystem dock Copy Path shortcut

This closes #43396.

KoBeWi committed:
Add append_array() method to Array class

rcorre committed:
Proposal 1246: Make Path3D handles more visible.

Resolves godotengine/godot-proposals#1246.

It is difficult to tell the difference between the handles for adjusting
curves and the points themselves when looking at a Path gizmo.
This re-uses the icons used for Path2D.

Unlike Path2D, this does not use a different icon for smooth vs sharp
points, as using a potentially different material for each point would
prevent batching the points in add_handles (and adding them out-of-order
messes up other logic based on handle indices).

This includes a public API change to allow specifying a texture for a
handle material. This allows spatial gizmo plugins to customize the way
a handle is rendered, if desired, but does not break existing behavior
(as providing no texture uses the default).

The path handle icons were resized as well.  16x16 is the standard icon
size. These icons were 10x10 rather than 16x16, and appeared rather
small in the editor.

To resize, I:

- Opened the original in Inkscape
- Resized the document to 16x16
- Opened the transform dialog
- Scaled by 160% proportionally
- Used Align/Distribute to center on the page
- Saved the document
- Cleaned with `svgcleaner --multipass`

reduz committed:
Refactored variant constructor logic

akien-mga committed:
Color: Fix -Wtype-limits GCC warning after refactoring

Warning from GCC 10.2.0 with `warnings=extra`:
```
./core/math/color.h: In member function 'int32_t Color::get_r8() const':
./core/typedefs.h:107:42: error: comparison of unsigned expression in '< 0' is always false [-Werror=type-limits]
  107 | #define CLAMP(m_a, m_min, m_max) (((m_a) < (m_min)) ? (m_min) : (((m_a) > (m_max)) ? m_max : m_a))
      |                                    ~~~~~~^~~~~~~~~
./core/math/color.h:201:49: note: in expansion of macro 'CLAMP'
  201 |  _FORCE_INLINE_ int32_t get_r8() const { return CLAMP(uint32_t(r * 255.0), 0, 255); }
      |                                                 ^~~~~
```

Also some code consistency changes while at it.

akien-mga committed:
CI: Build with warnings=extra to catch 'em all

... on all platforms but MSVC, as it still has a number of unsolved warnings
in its `/Wall` level. Some of it might be valid, others might be overkill,
this needs further assessment and fixes. (We could also change the `extra`
level to `/W4` on MSVC if that's more meaningful.)

bruvzg committed:
Update "open-simplex-noise-in-c" to fix undefined signed overflow.

akien-mga committed:
Variant: Rename Type::_RID to Type::RID

The underscore prefix was used to avoid the conflict between the `RID` class
name and the matching enum value in `Variant::Type`.

This can be fixed differently by prefixing uses of the `RID` class in `Variant`
with the scope resolution operator, as done already for `AABB`.

bruvzg committed:
Fix duplicate variant constructor typo.

akien-mga committed:
Variant: Sync docs with new constructors, fixups after #43403

Change DocData comparators for MethodDoc and ArgumentDoc to get a better
ordering of constructors.

reduz committed:
Make sure operators appear in the docs too

Add "operator" and "constructor" qualifiers to make it easier to
see in the docs.

reduz committed:
Change how no-arg constructor is handled internally in Variant.

Calinou committed:
Clarify that code in `assert()` should avoid side effects

GryphonClaw commented:
Can we please, please get an approval for a merge into 3.2.4?

akien-mga commented on editor/plugins/tile_set_editor_plugin.cpp:
```diff
@@ -500,6 +500,7 @@ TileSetEditor::TileSetEditor(EditorNode *p_editor) {
 	tools[SHAPE_NEW_RECTANGLE]->set_button_group(tg);
 	tools[SHAPE_NEW_RECTANGLE]->set_tooltip(TTR("Create a new rectangle."));
 	tools[SHAPE_NEW_RECTANGLE]->connect("pressed", callable_mp(this, &TileSetEditor::_on_tool_clicked), varray(SHAPE_NEW_RECTANGLE));
+	tools[SHAPE_NEW_RECTANGLE]->set_shortcut(ED_SHORTCUT("tileset_editor/editmode_collision/shape_new_rectangle", TTR("New Rectangle"), KEY_MASK_SHIFT | KEY_R));
```
Why are those under `editmode_collision`? It's not a section, it's the name of another shortcut.

GryphonClaw replied:
I think my logic was because it's in the edit mode collision editor that it should have a new section because that's different than the edit mode overall....

akien-mga replied:
I just tested and it seems to work, but it's a bit weird to have `tileset_editor/editmode_collision` be both an actual shortcut and a category for three other shortcuts.

Moreover those three buttons are used for the Collision, Occlusion and Navigation editors, so they're not specific to `editmode_collision` IMO.

I'd suggest using:
```
tileset_editor/shape_new_rectangle
tileset_editor/shape_new_polygon
tileset_editor/shape_delete
```

GryphonClaw replied:
Ok, changes submitted =)

Rubonnek committed:
Document InstancePlaceholder.create_instance not being thread-safe.

aaronfranke committed:
Allow getting Input axis/vector values by specifying multiple actions

For get_vector, use raw values and handle deadzones appropriately

HaSa1002 committed:
[Docs] Add C# example for EditorPlugin::forward_canvas_draw_over_viewport

Follow up to #43356

Faless committed:
[HTML5] AudioWorklet API implementation.

Rewrote AudioDriverJavaScript to support multiple processor nodes.
The old (and deprecated) ScriptProcessorNode when threads are not
available, and the new AudioWorklet API when threads are enabled.

The new implementation uses two ring buffers and a shared state to
communicated with the AudioWorklet thread.

The audio.worklet.js JavaScript file is always added to the export
template, but only really used (and downloaded) in the thread build.

Commit:
Updated gd_glue.cpp to work with the latest changes in the variant refactoring

Without this change the engine dont compile with the mono module enabled.

akien-mga committed:
doc: Fixups to #43419 which added operators in the docs

- Escape the method names as e.g. `operator <` is invalid XML.
- Add a hack to merge all String % operator definitions for each Variant type
  as a single one with `Variant` argument type.
- Add support for the new qualifiers in makerst.py.
- Drop unused `doc_merge.py`, seems to date back to when we had all the
  documentation in a single `classes.xml`.

akien-mga committed:
doc: Sync classref to add operators after #43419

qarmin committed:
Remove Attempted to free invalid ID error

Bhu1-V committed:
Checking for half_image_size[ x & y ] to be non-negative.

This make sure that (1x1) , (1 x X) and (X , 1) pixel images using sub-sampling will get correct half_image_size i.e NON-NEGATIVE.
fix : https://github.com/godotengine/godot/issues/42363

aaronfranke committed:
Minor clamp and float fixes

reduz committed:
Create Variant built-in functions.

-Moved Expression to use this, removed its own.
-Eventually GDScript/VisualScript/GDNative need to be moved to this.
-Given the JSON functions were hacked-in, removed them and created a new JSONParser class
-Made sure these functions appear properly in documentation, since they will be removed from GDScript

pouleyKetchoupp committed:
Expose LineEdit scroll offset to scripts

timothyqiu committed:
Keep cursor relative position after multiline move

naithar committed:
iOS: Implement pluggable application delegate.

reduz committed:
Refactor variant built-in methods yet again.

* Using C-style function pointers now, InternalMethod is gone.
* This ensures much better performance in typed code.
* Renamed builtin_funcs to utility_funcs, to avoid naming confusion

ccl2of4 committed:
Fix for linux joypad D-pad zeroing

Some controllers (notably those made by 8bitdo) do not always emit an event to zero out a D-pad axis before flipping direction. For example, when rolling around aggressively the D-pad of an 8bitdo SN30 Pro/Pro+, the following may be observed:

```
ABS_HAT0X : -1
ABS_HAT0Y : -1
ABS_HAT0Y : 0
ABS_HAT0Y : 1
ABS_HAT0X : 1
```

Notable here is that no event for `ABS_HAT0X: 0` is emitted between the events for `ABS_HAT0X: -1` and `ABS_HAT0X: 1`. Consequently, the game engine believes that both the negative _and_ positive x-axis directions of the D-pad are activated simultaneously (i.e `is_joy_button_pressed()` returns `true` for both `JOY_BUTTON_DPAD_LEFT` and `JOY_BUTTON_DPAD_RIGHT`), which should be impossible.

This issue is _not_ reproducible on all controllers. The Xbox One controller in particular will not exhibit this problem (it always emits zeroing out events for an axis before flipping direction).

The fix is to always zero out the opposite direction on the D-pad axis in question when processing an event with a nonzero value. This unfortunately wastes a small number of CPU cycles on controllers that behave nicely.

**I have verified this issue is also reproducible in the stable 3.2 branch**

pulawskig committed:
Removing unneeded FuncRef code in C#

fire committed:
Restore screenshot button.

bruvzg committed:
Add X11 Atom validity checks.

Calinou committed:
Use 75% editor scale on small displays automatically

This also makes borders always display in the editor theme,
even if the editor scale is below 100%. Otherwise, "focus" outlines
would vanish when using an editor scale below 100%,
which harms usability.

Commit:
Fix Android Export jarsigner error with *.import whitelist

Nickswoboda committed:
Improve error messages related to failing to open files

timothyqiu committed:
Allows HTTPClient to talk to proxy server

* Makes request uri accept absolute URL and authority
* Adds Host header only when missing

akien-mga committed:
SceneTree: Fix reparent crash with animation tracks renaming disabled

This check was there since the first commit in 2014, but a later feature added in 2018
with #17717 did not properly update the code while adding non animation-related code
in `perform_node_renames`.

Fixes #40532.

Maciej-Gorczyca committed:
Changed mouse cursor  to the caret (text cursor) location.

Calinou committed:
Clarify packet peer `max_buffer_po2` in ProjectSettings documentation

This closes https://github.com/godotengine/godot-docs/issues/4364.

m4gr3d committed:
Remove duplicate Android `orientation` settings.

Calinou committed:
Add a test suite for Curve

m4gr3d committed:
Clear unneeded assets when generating an apk expansion.
Clean up export wording to account for the different export formats.

mbrlabs committed:
Added the .jks file extension as valid preset for Android keystore files

m4gr3d committed:
Fix issue causing `Export all` to fail.

m4gr3d committed:
Add missing zip alignment step for generated apks.

bruvzg committed:
Fixes crash if Vulkan presentation surface is not available.

Calinou committed:
Improve the Dictionary class documentation

- Mention Lua-style syntax.
- Make the code samples self-contained.
- Mention caveat with `const` (also in Array).
- Clarify the description of `size()`.

This closes https://github.com/godotengine/godot-docs/issues/4272.

KoBeWi committed:
Correct the doc about Tree.get_edited

KoBeWi committed:
Allow folder checking in export preset file list

bruvzg committed:
Add XChangeProperty Atoms validity checks.

TokageItLab committed:
Lowering the Minimum Unit at Shape3D and Cameras

m4gr3d committed:
Update the logic to query for the 'scons' command executable path.

akien-mga committed:
Fix new transform to pixel snapping logic

Co-authored-by: lawnjelly <lawnjelly@gmail.com>

naithar committed:
GUI: send cursor data from TextEdit

naithar committed:
iOS: fix keyboard display

zaevi committed:
fix custom loader/saver broken

rcorre committed:
Clarify Curve3D.get_point_{in,out} position in doc.

I verified this experimentally. I added a point at roughly (1,0,0), and
dragged a handle back to the origin. The result was:

```
get_point_position: (0.991079, 0, -0.000069)
get_point_in: (0.993409, 0, 0)
get_point_out: (-0.993409, 0, 0)
```

Calinou committed:
Remove property groups for Pause Mode and Script

Each of those only grouped 1 property, making them useless.

This closes https://github.com/godotengine/godot-proposals/issues/1840.

Calinou committed:
Add a test suite for PCKPacker

Calinou committed:
Add a test suite for NodePath

mbrlabs committed:
Fixed exit code retrieval of spawned processes on Windows

Use GetExitCodeProcess() on Windows to retrieve the exit code of a process in
OS:excute()

RandomShaper committed:
Fix crash in resoure duplicate

akien-mga committed:
doc: Sync classref with Variant utility methods

Copied relevant documentation from the original `@GDScript` built-ins,
which will likely be removed in a future commit.

Various fixups to `variant_utility.cpp` while working on this.

akien-mga committed:
i18n: Sync translations with Weblate

(cherry picked from commit e95af7ae9b883151e6de1d31f94950833ae5b478)

akien-mga committed:
Debugger: Save options in project metadata

Fixes #19542.

Faless committed:
Fix CLI export when export_path is in preset.

Export presets contains the export_path option, to specify the default
export location, but the CLI export option disregarded that, and always
required and export path to be specified.
After this commit, if the export path is not specified in the command,
the one in the preset will be used, erroring only if it's not present or
invalid.

madmiraal committed:
Fix godot_js_websocket_send function unused assignment.

madmiraal committed:
Add missing javascript semi-colons.

akien-mga commented:
Could you squash the two commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

mbrlabs committed:
Fixed deletion of nodes with exported node paths

akien-mga committed:
nanosvg: Sync with upstream cc6c08d

Includes some loop condition fixes after fuzzing.

timothyqiu committed:
Fixes rotation in select mode on macOS

akien-mga committed:
glTF: Workaround import failure with invalid embedded images

image/gif is not supported in the glTF 2.0 specification,
these files are broken. But let's be lenient...

Fixes #43638.

akien-mga committed:
Revert "nanosvg: Sync with upstream cc6c08d"

This reverts commit f697e7879b0c1b37b94b890b3f2a9910e5d822a2.

Part of the update introduced a regression:
https://github.com/memononen/nanosvg/issues/188.

We could include a local revert of the problematic commit but let's just do a
full revert to our previous version, and I'll re-update once the regression
is fixed upstream.

Fixes #43641.

vnen committed:
Fix return type on builtin validated calls

Before it was being set to the base type instead of the actual return
type.

jordo committed:
allow vulkan validation layers in release builds if explicity asked for

qarmin committed:
Allow to open multiple projects when some are not imported or without main scene.

GryphonClaw commented:
> Could you squash the two commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

sorry i dont have the patience or understanding how to do that, i _barely_ got this to work in the first place.

GryphonClaw commented:
closing to improve the PR

Calinou commented:
Superseded by https://github.com/godotengine/godot/pull/43668.
